File: DAL/Entities/Asset.cs
namespace Business
{
    using System;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    public partial class Asset
    {
        [Key]
        [DatabaseGeneratedAttribute(DatabaseGeneratedOption.Identity)]
        //[DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int Id { get; set; }

        [StringLength(50)]
        public string Description { get; set; }

        public int? EmployeeId { get; set; }

        public int? AssetCategoryId { get; set; }

        public int? StatusId { get; set; }

        public int? DepartmentID { get; set; }

        [StringLength(50)]
        public string Make { get; set; }

        [StringLength(15)]
        public string ModelNumber { get; set; }

        [StringLength(50)]
        public string ComputerName { get; set; }

        public DateTime? DateAcquired { get; set; }
    }
}

File: AssetCoreSol/ClientApp/src/app/models/asset.ts
import { Category } from "./category";

 export class Asset {

  id: string;
  description: string;
  employeeId: string;
  assetCategoryId: string;
  statusId: string;
  departmentID: string;
  make: string;
  modelNumber: string;
  computerName: string;
  dateAcquired: string;

  //mapped names
  statusName: string;
  categoryName: string;
  departmentName: string;
  employeeName: string;
}
